polytracker/src/taint_sources/argv.cpp

```diff
@@ -0,0 +1,32 @@
+#include <string>
+
+#include "polytracker/early_construct.h"
+#include "polytracker/polytracker.h"
+#include "taintdag/polytracker.h"
+
+EARLY_CONSTRUCT_EXTERN_GETTER(taintdag::PolyTracker, polytracker_tdag);
+
+namespace polytracker {
+
+void taint_argv(int argc, char *argv[]) {
+
+  // The check could be done in the calling code, for performance reasons.
+  // However this function should only ever be invoked once (from main).
+  if (!polytracker_taint_argv)
+    return;
+
+  if (argc <= 0) {
+    // Weird. Not much to do though.
+    return;
+  }
+
+  auto &polyt = get_polytracker_tdag();
+
+  for (int i = 0; i < argc; i++) {
```

Comment:
nitpick: I think `++i` are generally preferred unless there's a reason for a `i++`.